pkg/clusteragent/custommetrics/provider.go

```diff
@@ -73,6 +73,10 @@ func (p *datadogProvider) ListAllExternalMetrics() []provider.ExternalMetricInfo
 	rawMetrics := p.hpaClient.ReadConfigMap()
 
 	for _, metric := range rawMetrics {
+		// Only metrics that exist in Datadog and available are eligible to be evaluated in the HPA process.
+		if !metric.Valid {
+			continue
+		}
```

Comment:
Would it make sense to log each time `ListAllExternalMetrics` and `GetExternalMetric` is called at the `Debug` level with the data we are returning?

Reply:
It's quite verbose but I'll add a Trace level log in the ListAll